pkg/hostagent/setup.go

```diff
@@ -284,6 +302,18 @@ func (agent *HostAgent) configureContainerIfaces(metadata *md.ContainerMetadata)
 			}
 		}
 
+		for _, ip := range iface.IPs {
+			if ip.Address.IP != nil {
```

Comment:
In the dual stack case, if you see the v6 first, then will we not derive the MAC address based on v4 IP?

Reply:
Yes, Thats right. We yet don't support IPv6 isn't it ?

Reply:
In the dual stack case, the MAC address needs to be derived based on ipv4. Will that happen here if the ipv6 address is processed first (since you break out of the loop once the ipv6 is processed)?

Reply:
My understanding was that the IPv4 address will always come first in the dual-stack case. That's based on mailing threads I've seen in the past though and I'm not 100% sure it's guaranteed

Reply:
To avoid any confusions, I will have a check on ipv4 and on ipv6. If Ipv6, I will continue with the for loop and if ipv4, I will call runSetupVeth.

Reply:
That would break if we ever need to support PODs with only IPv6 addresses

Reply:
How about handling ipv4 by assigning mac address based on IP, and for ipv6 retain the current implementation (Once the upstream supports ipv6 based mac allocation, we can handle it by calling ip based mac alloc)   ?

Reply:
Sure that's fine

Reply:
ok thanks Tom

Reply:
so 4 cases: 
1. ipv4 only
2. ipv6 only
3. ipv6 and ipv4 
4. ipv4 and ip6 

step1. if ipv4 only, check for ipv4 -> call runSetupVeth -> assign mac based on ipv4 ->  assign the results i.e result.hostveth and result.mac 
step2. if ipv6 only, check for ipv6 > call runSetupVeth -> “NO” mac based ip assignment > follow the current implementation
step3: for ipv6 and ipv4 , if ipv6 comes first > repeat step2 > for the next iteration, check if veth pair already exists. if yes, break the loop (this is to avoid creating two veths for dual stack setup)
step4: for ipv4 and ipv6, if ipv4 comes first -> repeat step1 >   for the next iteration, check if veth pair already exists. if yes, break the loop

Reply:
I will proceed with this logic if no concerns